realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -271,13 +271,13 @@ public void refresh() {
         if (isInTransaction()) {
             throw new IllegalStateException(BaseRealm.CANNOT_REFRESH_INSIDE_OF_TRANSACTION_MESSAGE);
         }
-        sharedGroupManager.advanceRead();
-        if (handlerController != null) {
-            handlerController.notifyAllListeners();
-            // if we have empty async RealmObject then rerun
-            if (handlerController.threadContainsAsyncEmptyRealmObject()) {
-                handlerController.updateAsyncEmptyRealmObject();
-            }
+        if (!handlerController.isAutoRefreshEnabled()) {
+            // non Looper Thread, just advance the Realm
+            // registering listeners is not allowed, hence nothing to notify
+            sharedGroupManager.advanceRead();
+            handlerController.refreshTableViews();
```

Comment:
This code is only triggered if there is no auto-refresh, so there isn't any async queries either